handlers/rsa.go

```diff
@@ -0,0 +1,222 @@
+package handlers
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"sync"
+
+	"github.com/vocdoni/blind-csp/csp"
+	"go.vocdoni.io/dvote/db"
+	"go.vocdoni.io/dvote/db/metadb"
+	"go.vocdoni.io/dvote/log"
+)
+
+const (
+	ELECTION_ID_STR_LENGTH      = 64
+	VOTER_ID_STR_LENGTH         = 64
+	SIGNED_MESSAGE_BYTES_LENGTH = 64
+)
+
+var rsaPubKeys = []string{
+	`-----BEGIN PUBLIC KEY-----
+MIIBIDANBgkqhkiG9w0BAQEFAAOCAQ0AMIIBCAKCAQEApc2hU8zulyJzdQE5IPAv
+B2BgveoZmYUmPEjSb4DViBoATK1hlaY8Psp5vj0H0L4tM8AlXRhPQlECibhgccig
+xQFcG7CLXiSAn7c4XoR+J2SCgx76Fwl9L3WhQigxyKsmpGIqubseydmwfJi4TBnq
+qnX4prsW1PT8GpG35t8Qi8PtkXVGmL7G5pkPXtF0hRzKSfhzsDBbJsl6Jk/Rn5Id
+pKHXL22FdbE9fGzIlW2a6Zdd0b0Q3FZBMnWLSwo0OwBtC/qNnDTCzboig9djiFmA
+yuj8jVhsy050nI72TAONjGKi+xn4lYfdOV2k6TyvpRHfylHouK2v0/bktSlkFI0y
+nwIBAw==
+-----END PUBLIC KEY-----`,
+}
+
+// RsaHandler is a handler that allows only 1 registration for IP
+type RsaHandler struct {
+	kv       db.Database
+	keysLock sync.RWMutex
+}
+
+func (rh *RsaHandler) addKey(index, value []byte) error {
+	rh.keysLock.Lock()
+	defer rh.keysLock.Unlock()
+	tx := rh.kv.WriteTx()
+	defer tx.Discard()
+	if err := tx.Set(index, value); err != nil {
+		return err
+	}
+	return tx.Commit()
+}
+
+func (rh *RsaHandler) exist(index []byte) bool {
+	rh.keysLock.RLock()
+	defer rh.keysLock.RUnlock()
+	tx := rh.kv.WriteTx()
+	defer tx.Discard()
+	_, err := tx.Get(index)
+	return err == nil
+}
+
+// GetName returns the name of the handler
+func (rh *RsaHandler) GetName() string {
+	return "rsa"
+}
+
+// Init initializes the handler.
+// Takes one argument for persistent data directory.
+func (rh *RsaHandler) Init(opts ...string) (err error) {
+	rh.kv, err = metadb.New(db.TypePebble, filepath.Clean(opts[0]))
```

Comment:
The first argument of opts is the "dataDir" introduced in the command line flag. The second argument provided should be the RSA key file for validating the queries.

So instead of hardcoding the RSA key (as done in line 27) I'd use the Init options array for including the key. This will make the handler generic for any use.